### PR TITLE
Feat/1xxx maui service scope 2

### PR DESCRIPTION
**Pull request by crui3er**
In MAUI app enables contaner services scope validation in Debug mode only.
Fixes some service registrations.

### CODE REVIEW

File: src/dotnet/Core/DependencyInjection/ServiceCollectionExt.cs [hunk 1]
```diff
@@ -26,4 +26,74 @@ public static class ServiceCollectionExt
...
         => services.AddSingleton(_ => new TypeMap<TScope>(typeMapBuilder));
+
+    public static ServiceDescriptor ChangeLifetime(
+        this ServiceDescriptor serviceDescriptor,
+        ServiceLifetime serviceLifetime)
+    {
+        if (serviceDescriptor.ImplementationFactory != null)
+            return new ServiceDescriptor(serviceDescriptor.ServiceType,
+                serviceDescriptor.ImplementationFactory,
+                serviceLifetime);
```
Comment (alexis-kochetov) [inline src/dotnet/Core/DependencyInjection/ServiceCollectionExt.cs:37]: please reformat with Rider and add new lines

File: src/dotnet/Core/Performance/ServiceCollectionExt.cs [hunk 2]
```diff
@@ -2,13 +2,18 @@ namespace ActualChat.Performance;
...
 {
-    public static IServiceCollection AddTracer(this IServiceCollection services, Tracer? tracer = null)
+    public static IServiceCollection AddTracer(this IServiceCollection services)
     {
-        if (tracer == null)
-            services.AddScoped(_ => new ScopedTracerProvider());
-        else
-            services.AddScoped(_ => new ScopedTracerProvider(tracer));
+        if (services.Any(c => c.ServiceType == typeof(Tracer)))
+            return services;
```
Comment (alexis-kochetov) [inline src/dotnet/Core/Performance/ServiceCollectionExt.cs:8]: Add newline

File: src/dotnet/Core/DependencyInjection/ServiceCollectionExt.cs (hunk 1 above)
Comment (crui3er) [reply src/dotnet/Core/DependencyInjection/ServiceCollectionExt.cs:37]: done

File: src/dotnet/Core/Performance/ServiceCollectionExt.cs (hunk 2 above)
Comment (crui3er) [reply src/dotnet/Core/Performance/ServiceCollectionExt.cs:8]: done

**Pull request merged**